File: backend/app/models/product.py
from sqlalchemy import Column, Integer, String, Float
from app.db.database import Base


class Product(Base):
    __tablename__ = "products"

    id = Column(Integer, primary_key=True, index=True)
    product_name = Column(String, nullable=False)
    category = Column(String)
    brand = Column(String)
    size = Column(String)
    color = Column(String)
    price = Column(Float)
    quantity = Column(Integer)

File: frontend/src/types/product.ts
export interface Product {
  id: number;
  product_name: string;
  category?: string;
  brand?: string;
  size?: string;
  color?: string;
  price: number;
  quantity: number;
  image_url?: string;
}

export interface ProductListResponse {
  total: number;
  page: number;
  limit: number;
  total_pages: number;
  data: Product[];
}